backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/document/model/impl/LuceneIndexSchemaFieldNode.java

```diff
@@ -50,6 +55,18 @@ public LuceneFieldQueryFactory getQueryFactory() {
 		return queryFactory;
 	}
 
+	public LuceneFieldSortContributor getSortContributor() {
+		return sortContributor;
+	}
+
+	public boolean isCompatibleWith(LuceneIndexSchemaFieldNode<?> other) {
+		if ( !Objects.equals( sortContributor, other.sortContributor ) || !Objects.equals( formatter, other.formatter )
+				|| !Objects.equals( queryFactory, other.queryFactory ) ) {
+			return false;
+		}
+		return true;
+	}
+
 	@Override
 	public String toString() {
 		return getClass().getSimpleName() + "[parent=" + parent + ", fieldName=" + fieldName + "]";
```

Comment:
Since you are using this `toString()` implementation in logs to explain formatting conflicts, it should probably include some information about the format.